### ISSUE TITLE
Usage reports not working in Openstack 

### ISSUE THREAD

**Issue by lwr20** [projectcalico/calico#1099]
I've set up centos/liberty openstack system using packages build from master to investigate why we aren't getting usage reports from this sort of system.  felix.log showed "Exception in usage_report" errors.  After modifying futils.py to change the line which logs that error from 
`_log.info("Exception in usage_report")` to `_log.exception("Exception in usage_report")`, I see the stacktrace looks like this:

```
2016-09-14 12:37:14,741 [ERROR][17205/29] calico.felix.futils 672: Exception in usage_report
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/calico/felix/futils.py", line 668, in report_usage_and_get_warnings
    r = http.request('GET', url, fields=fields, retries=retries)
  File "/usr/lib/python2.7/site-packages/urllib3/request.py", line 69, in request
    **urlopen_kw)
  File "/usr/lib/python2.7/site-packages/urllib3/request.py", line 90, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/usr/lib/python2.7/site-packages/urllib3/poolmanager.py", line 165, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/usr/lib/python2.7/site-packages/urllib3/connectionpool.py", line 578, in urlopen
    chunked=chunked)
  File "/usr/lib/python2.7/site-packages/urllib3/connectionpool.py", line 351, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python2.7/site-packages/urllib3/connectionpool.py", line 814, in _validate_conn
    conn.connect()
  File "/usr/lib/python2.7/site-packages/urllib3/connection.py", line 289, in connect
    ssl_version=resolved_ssl_version)
  File "/usr/lib/python2.7/site-packages/urllib3/util/ssl_.py", line 308, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib64/python2.7/ssl.py", line 350, in wrap_socket
    _context=self)
TypeError: __init__() got an unexpected keyword argument 'server_hostname'
2016-09-14 12:37:14,743 [INFO][17205/29] calico.felix.fetcd 387: Usage report interval: 86400, sleep-time: 83180.2611056
```
@fasaxc suggests it may be some sort of version incompatibility in urllib3.  

```
$yum list installed | grep urllib3
python-urllib3.noarch                1.15.1-2.el7             @openstack-liberty
```

**Comment by neiljerram** [comment]
https://github.com/shazow/urllib3/issues/482, apparently (via https://github.com/asciimoo/searx/issues/120)

**Comment by neiljerram** [comment]
Actually https://github.com/gevent/gevent/issues/477 is the root problem.

**Comment by neiljerram** [comment]
Fixed by #1103
